source/include/sigv4_config_defaults.h

```diff
@@ -77,6 +77,19 @@
     #define SIGV4_MAX_QUERY_PAIR_COUNT    100U
 #endif
 
+/**
+ * @brief Macro used to compute the worst-case stack size when sorting elements
+ * associated with #SIGV4_MAX_QUERY_PAIR_COUNT or #SIGV4_MAX_HTTP_HEADER_COUNT.
+ * Suppose the max of the two aforementioned macros is X, then the macro
+ * below must be set to ceiling( log(X)/log(2) ) where ceiling rounds up
+ * the ones digit if the decimal is greater than 0.
+ * @note If updating #SIGV4_MAX_QUERY_PAIR_COUNT or #SIGV4_MAX_HTTP_HEADER_COUNT,
+ * be sure to update this value based on the formula above.
+ */
+#ifndef SIGV4_WORST_CASE_SORT_STACK_SIZE
+    #define SIGV4_WORST_CASE_SORT_STACK_SIZE    7U
```

Comment:
Comparing this with glibc, I think this value needs to be doubled. They use one stack entry per pair of low/high pointers, while this one uses two entries